### PR TITLE
Add accept-post header model

### PR DESCRIPTION
**Pull request by coltfred**
Addresses part of #5010

### CODE REVIEW

**Review by rossabaker — CHANGES_REQUESTED** [review]
Looks good after the list tweak.

File: core/src/main/scala/org/http4s/headers/Accept-Post.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    headerRep1(MediaType.parser).map(xs => `Accept-Post`(xs.head, xs.tail: _*))
+
+  implicit val headerInstance: Header[`Accept-Post`, Header.Recurring] =
+    Header.createRendered(
+      ci"Accept-Post",
+      _.values,
+      parse
+    )
+
+  implicit val headerSemigroupInstance: cats.Semigroup[`Accept-Post`] =
```
Comment (rossabaker) [inline core/src/main/scala/org/http4s/headers/Accept-Post.scala:43]: Well, heck, you probably can even have a Monoid after the other comment!

Comment (coltfred) [reply core/src/main/scala/org/http4s/headers/Accept-Post.scala:43]: Can I assume there should be a comment about it maybe being able to just be a List inside?

I made it a Nel to match[ the Accept header](https://github.com/http4s/http4s/blob/series/0.22/core/src/main/scala/org/http4s/headers/Accept.scala#L81)

Comment (rossabaker) [reply core/src/main/scala/org/http4s/headers/Accept-Post.scala:43]: The `Accept` header is wrong, too.  I think @bplommer fixed a couple of the CORS headers that made this mistake on 1.0, and you might find precedent there.

Comment (coltfred) [reply core/src/main/scala/org/http4s/headers/Accept-Post.scala]: @rossabaker I ported the changes made by @bplommer to the series/0.22. I tried to port it into the exact lines to make potential merges less painful. Let me know if there's any other changes you'd like to see.

File: core/src/main/scala/org/http4s/internal/parsing/Rfc7230.scala [hunk 2]
```diff
@@ -77,6 +77,9 @@ private[http4s] object Rfc7230 {
     between(char('('), cText.orElse(quotedPair).orElse(comment).rep0.string, char(')'))
   }
 
+  def headerRep[A](element: Parser[A]): Parser0[List[A]] =
```
Comment (rossabaker) [inline core/src/main/scala/org/http4s/internal/parsing/Rfc7230.scala:80]: I am surprised this didn't exist already, but I can't find it elsewhere.

**Review by rossabaker — APPROVED** [review]

**Pull request merged**